pkg/messaging/v1/util.go

```diff
@@ -148,14 +180,33 @@ func reservedGRPCMetadataToDaprPrefixHeader(key string) string {
 }
 
 // InternalMetadataToHTTPHeader converts internal metadata pb to HTTP headers
-func InternalMetadataToHTTPHeader(internalMD DaprInternalMetadata, setHeader func(string, string)) {
+func InternalMetadataToHTTPHeader(ctx context.Context, internalMD DaprInternalMetadata, setHeader func(string, string)) {
+	var traceparentValue, tracestateValue, grpctracebinValue string
 	for k, listVal := range internalMD {
-		// Skip if the header key has -bin suffix
-		if len(listVal.Values) == 0 || strings.HasSuffix(k, gRPCBinaryMetadataSuffix) || k == ContentTypeHeader || isTraceCorrleationHeaderKey(k) {
+		// get both the trace headers for HTTP/GRPC and continue
+		if isTraceParentHeaderKey(k) {
+			traceparentValue = listVal.Values[0]
+			continue
+		}
+		if isTraceStateHeaderKey(k) {
+			tracestateValue = listVal.Values[0]
+			continue
+		}
+		if isGRPCTraceHeaderKey(k) {
+			grpctracebinValue = listVal.Values[0]
+			continue
+		}
```

Comment:
In this case, switch-case is simpler than helper method + multiple ifs.

switch strings.ToLower(k) {
case 
}

Reply:
done.